Task1 :

Q1: You survey households in your area to find the average rent they are paying. Find the standard deviation from the following data: $1550, $1700, $900, $850, $1000, $950.

Q2: Find the variance for the following set of data representing trees in California (heights in feet): 3, 21, 98, 203, 17, 9

Q3: In a class on 100 students, 80 students passed in all subjects, 10 failed in one subject, 7 failed in two subjects and 3 failed in three subjects. Find the probability distribution of the variable for number of subjects a student from the given class has failed in.

In [1]:
import numpy as np

Q1: You survey households in your area to find the average rent they are paying. Find the standard deviation from the following data: $1550, $1700, $900, $850, $1000, $950.

In [2]:
x = [1550,1550,1700, 900,900,850, 1000,1000,950]
Avg_rent_paid = np.mean(x)
std_dev = np.std(x)

print('Average rent paid by the householdsfor the sample information collected :  {:0.2f}'.format(Avg_rent_paid))
print('The limits of data pertaining to rent paid by the houesholds is +/- : {:0.2f}'.format(std_dev))

Average rent paid by the householdsfor the sample information collected :  1155.56
The limits of data pertaining to rent paid by the houesholds is +/- : 320.11


Average rent paid by the householdsfor the sample information collected :  1155.56
The limits of data pertaining to rent paid by the houesholds is +/- : 320.11
Q2:Find the variance for the following set of data representing trees in California (heights in feet): 3, 21, 98, 203, 17, 9

In [3]:
#x : height of trees in feet
x = [3, 21, 98, 203, 17, 9]
variance = np.var(x)

print("The spread of data pertaining to the height of trees is : {:0.2f}".format(variance))

The spread of data pertaining to the height of trees is : 5183.25


The spread of data pertaining to the height of trees is : 5183.25
Q3:In a class on 100 students, 80 students passed in all subjects, 10 failed in one subject, 7 failed in two subjects and 3 failed in three subjects. Find the probability distribution of the variable for number of subjects a student from the given class has failed in.

In [4]:
import matplotlib.pyplot as plt
%matplotlib inline
n_students = 100
passed_all = 80
failed_one = 10
failed_two = 7
failed_three = 3

num_students_not_passed_all = n_students - passed_all
prob_failed_none = passed_all / n_students
prob_failed_in_one = failed_one/n_students
prob_failed_in_two = failed_two/n_students
prob_failed_in_three = failed_three/n_students

print("Probability failed in no subjects: ",prob_failed_none)
print("Probability failed in 1 subject: ",prob_failed_in_one)
print("Probability failed in 2 subjects: ",prob_failed_in_two)
print("Probability failed in 3 subjects: ",prob_failed_in_three)

Probability failed in no subjects:  0.8
Probability failed in 1 subject:  0.1
Probability failed in 2 subjects:  0.07
Probability failed in 3 subjects:  0.03
